### ISSUE TITLE
Cross platform makefile needed

### ISSUE THREAD

**Issue by wornwinter** [wornwinter/DMGe#33]
I think we need a cross platform makefile and possibly a configure script.

I'm finding that moving between different systems requires too much customisation of the code::blocks project.

Building is becoming a bit of a chore.

**Comment by Quaker762** [comment]
Yeah sure that sounds good. What build stuff would you use for Windows then? CMake??

**Comment by wornwinter** [comment]
Haven't really considered yet. Whatever you can make work will suit; I'm not working much on Windows nowadays.

On another note, there are some issues that need resolved with Linux. It segfaults when loaded and also causes X11 errors (something to do with the GL canvas). I'll hopefully do some work on it over the next few days.

At least it builds now. Took me about an hour to just get the thing to build on Linux last night lol.

**Comment by Quaker762** [comment]
Okay cool, I'll get working on a Windows build environment. Should the dependencies be located in the DMGe folder??

**Comment by wornwinter** [comment]
Ideally we should be able to configure it so that it does not matter where
the dependencies are stored. See the Linux makefile where I'm using
'wx-config' etc?

**Comment by Quaker762** [comment]
Okay that can be done in Cmake. I nearly got a makefile for it done, but the syntax is a bit convoluted. I should have it done soon though.
